pkg/apply/path.go

```diff
@@ -0,0 +1,109 @@
+package apply
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+func processPaths(paths []string) (genericclioptions.FileNameFlags, error) {
+	// No arguments means we are reading from StdIn
+	fileNameFlags := genericclioptions.FileNameFlags{}
+	if len(paths) == 0 {
+		fileNames := []string{"-"}
+		fileNameFlags.Filenames = &fileNames
+		return fileNameFlags, nil
+	}
+
+	// If there are more than one argument, they must all be files. This also
+	// means they can't contain Kustomizations
+	t := true
+	if len(paths) > 1 {
+		// Check that all paths point to a valid file.
+		var invalidPaths []string
+		for _, path := range paths {
+			isFile, err := isFile(path)
+			if err != nil {
+				if os.IsNotExist(err) {
+					invalidPaths = append(invalidPaths, path)
+					continue
+				}
+				return fileNameFlags, err
+			}
+			if !isFile {
+				invalidPaths = append(invalidPaths, path)
+			}
+		}
+		if len(invalidPaths) > 0 {
+			return fileNameFlags, fmt.Errorf("invalid paths: %s", strings.Join(invalidPaths, ","))
+		}
+		fileNameFlags.Filenames = &paths
+		fileNameFlags.Recursive = &t
+		return fileNameFlags, nil
+	}
+
+	// If we have exactly one parameters, we check if the target
+	// is a Kustomization.
+	isKustomization, err := hasKustomization(paths[0])
```

Comment:
personally i think that the apply operation should not have to know about kustomize.
use cases we'll support:
 - pipe from kustomize to kapply
 - pipe from kpt to kapply
 - or more likely, apply directly from kpt and/or kustomize.

@pwittrock ?